bee-ledger/src/types/output_diff.rs

```diff
@@ -3,14 +3,25 @@
 
 use crate::types::{error::Error, TreasuryDiff};
 
-use bee_common::packable::{Packable, Read, Write};
 use bee_message::output::OutputId;
+use bee_packable::prefix::{UnpackPrefixError, VecPrefix};
+
+use core::convert::Infallible;
+
+fn unpack_prefix_error_to_error(err: UnpackPrefixError<bee_message::Error, Infallible>) -> Error {
+    match err {
+        UnpackPrefixError::Packable(err) => err.into(),
+        UnpackPrefixError::Prefix(err) => match err {},
+    }
+}
 
 /// A type to record output and treasury changes that happened within a milestone.
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq, bee_packable::Packable)]
+#[packable(unpack_error = Error, with = unpack_prefix_error_to_error)]
 pub struct OutputDiff {
-    created_outputs: Vec<OutputId>,
-    consumed_outputs: Vec<OutputId>,
+    created_outputs: VecPrefix<OutputId, u32>,
```

Comment:
Do we really have to use a VecPrefix ?

Reply:
There is no prefix ? Should we always assume one ? Idk I feel like I would leave it panic

Reply:
I think I'm not following, the prefix is `u32` isn't it?

Reply:
Ah yeah my bad I confused myself. I meant we have no specific bound, It's sad that it leaks to `new`... Not sure I wouldn't unwrap there tbh...

Reply:
Yeah it is a hard to get error but even then `OutputDiff::new` is only called once in `apply_milestone`.